addons/external_intercom/functions/fnc_disconnect.sqf

```diff
@@ -0,0 +1,50 @@
+#include "script_component.hpp"
+
+/*
+  Name: TFAR_external_intercom_fnc_disconnect
+
+  Author: Arend(Saborknight)
+    Disconnects the player from the intercom of a vehicle.
+
+  Arguments:
+    0: Vehicle object <OBJECT>
+    1: Player unit <OBJECT>
+
+  Return Value:
+    None
+
+  Example:
+    [_vehicle, _player] call TFAR_external_intercom_fnc_disconnect;
+
+  Public: Yes
+*/
+
+params ["_vehicle", "_player"];
+
+["ace_unconscious", _player getVariable ["TFAR_ExternalIntercomEHID", -1]] call CBA_fnc_removeEventHandler;
+_player setVariable ["TFAR_vehicleIDOverride", nil, true];
+_player setVariable ["TFAR_ExternalIntercomVehicle", nil, true];
+
+_externalIntercomSpeakers = _vehicle getVariable ["TFAR_ExternalIntercomSpeakers", [objNull, []]];
+
+if ((_externalIntercomSpeakers select 0) isEqualTo _player) then {
+    _externalIntercomSpeakers set [0, objNull];
+};
+(_externalIntercomSpeakers select 1) deleteAt ((_externalIntercomSpeakers select 1) find _player);
+
+_vehicle setVariable ["TFAR_ExternalIntercomSpeakers", _externalIntercomSpeakers, true];
```

Comment:
potential race condition if two players disconnect at the same time in MP.
player1 might re-add player2 to the array even after player2 disconnected.
Not sure how to solve this, and also probably not that significant

Reply:
I'm not sure how to deal with the possible race condition either... maybe someone else will have an epiphane for us.

Reply:
Let the server handle it.
Like with radio ID's. Tell the server "hey that happened" and the server takes care of checking and doing.
The server cannot do two things in parallel.

Reply:
boop still open, though unlikely to happen. But maybe add a comment that thats possible and could potentially be fixed in the future